### DISCUSSION TITLE
Return table from SQL query when using as framework

### DISCUSSION THREAD

**Discussion by SethBurkart123** [pocketbase/pocketbase#1299] [Q&A] [answered]
Hi there,

I am trying to use PocketBase as a framework for my project and I am wondering how I can return the users table from the SQLite database. Is there a specific method or function that I should be using to access this table? I have tried using the following code but instead of returning the users it only returns:
```json
{
  "Locker": {}
}
```

```go
app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
	e.Router.AddRoute(echo.Route{
		Method: http.MethodGet,
		Path:   "/test",
		Handler: func(c echo.Context) error {
			request := app.Dao().DB().
				Select("*").
				From("users")
			result, err := request.Build().Execute()

			if err != nil {
				return c.JSON(http.StatusInternalServerError, err)
			}

			return c.JSON(http.StatusOK, result)
		},
		Middlewares: []echo.MiddlewareFunc{
			apis.ActivityLogger(app),
		},
	})

	return nil
})
```

I realise there was a similar question asked but couldn't get it to work with user data. I'd really like to return a list of user `id`'s from the `users` collection.
Any help would be greatly appreciated. Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
If you just want to get a list with the `users` records id, then you can use something like the following:

```go
var ids []string
err := app.Dao().DB().Select("id").From("users").Column(&ids)
```

`app.Dao().DB()` returns a struct from the dbx package. You can find more information and examples of the available methods in https://github.com/pocketbase/dbx#readme.

But if you want to return a list of `users` records (not just their ids), then I recommend using the `Dao` helpers instead. You can find some examples in https://pocketbase.io/docs/record-methods/#fetch-records.
